azure/functions/decorators/function_app.py

```diff
@@ -3297,8 +3287,27 @@ def get_functions(self) -> List[Function]:
                 '-bindings-http-webhook-trigger?tabs=in-process'
                 '%2Cfunctionsv2&pivots=programming-language-python#http-auth')
 
+        FunctionRegister.validate_functions(self, functions=functions)
+
         return functions
 
+    def validate_functions(self, functions: List[Function]) -> bool:
+        """The functions_bindings dict contains the function name and
+        its bindings for all functions in an app. If a previous function
+        has the same name, indexing will fail here.
+        """
+        if not self.functions_bindings:
+            self.functions_bindings = {}
+        for function in functions:
+            function_name = function.get_function_name()
+            if self.functions_bindings.get(function_name, None):
```

Comment:
We don't need to get the value of the function_name right? Can be changed to

```suggestion
            if function_name in self.functions_bindings:
```

Reply:
Dicts aren't iterable. If I changed it to a list that would work, but it would be N time complexity, which is why I made it a dict

Reply:
I didn't get it. We are not changing it to a list.. We can just check if the function name is a key in bindings dictionary